types/validator_set.go

```diff
@@ -131,11 +139,34 @@ func (vals *ValidatorSet) computeAvgProposerPriority() int64 {
 	panic(fmt.Sprintf("Cannot represent avg ProposerPriority as an int64 %v", avg))
 }
 
-func (vals *ValidatorSet) shiftByAvgProposerPriority(validatorsHeap *cmn.Heap) {
+// compute the difference between the max and min ProposerPriority of that set
+func computeMaxMinPriorityDiff(vals *ValidatorSet) int64 {
+	max := int64(math.MinInt64)
+	min := int64(math.MaxInt64)
+	for _, v := range vals.Validators {
+		if v.ProposerPriority < min {
+			min = v.ProposerPriority
+		}
+		if v.ProposerPriority > max {
+			max = v.ProposerPriority
+		}
+	}
+	diff := max - min
+	return diff
+}
+
+func (vals *ValidatorSet) getValWitMostPriority() *Validator {
+	var res *Validator
+	for _, val := range vals.Validators {
+		res = res.CompareProposerPriority(val)
```

Comment:
This is pretty unintuitive to read - do most objects in Tendermint have functions called `CompareXYZ` which return the higher-valued object by some property? Maybe I'm just not used to it. Otherwise I think directly comparing `val.ProposerPriority` or having a function `LargestValidatorByPriority` would be a bit more obvious.

Reply:
It is the same method as used in `GetProposer` (via `findProposer`). I agree that it is weird, though.